contracts/rust/deployer/src/network_config.rs

```diff
@@ -0,0 +1,119 @@
+//! Fetching hotshot network config
+
+use std::time::Duration;
+
+use alloy::{primitives::U256, transports::http::reqwest::Url};
+use anyhow::{Context, Result};
+use espresso_types::{config::PublicNetworkConfig, SeqTypes};
+use hotshot_contract_adapter::{
+    field_to_u256,
+    sol_types::{LightClientStateSol, StakeTableStateSol},
+};
+use hotshot_types::{
+    light_client::compute_stake_table_commitment,
+    traits::node_implementation::{ConsensusTime, NodeType},
+    PeerConfig,
+};
+use tokio::time::sleep;
+use vbs::version::StaticVersion;
+
+/// Returns both genesis light client state and stake table state
+pub async fn light_client_genesis(
+    sequencer_url: &Url,
+    stake_table_capacity: usize,
+) -> anyhow::Result<(LightClientStateSol, StakeTableStateSol)> {
+    let st = fetch_stake_table_from_sequencer(sequencer_url, None)
+        .await
+        .with_context(|| "Failed to initialize stake table")?;
+    light_client_genesis_from_stake_table(&st, stake_table_capacity)
+}
+
+/// Fetch the stake table from a sequencer node given the epoch number
+///
+/// Does not error, runs until the stake table is provided.
+pub async fn fetch_stake_table_from_sequencer(
+    sequencer_url: &Url,
+    epoch: Option<<SeqTypes as NodeType>::Epoch>,
+) -> Result<Vec<PeerConfig<SeqTypes>>> {
+    tracing::info!("Initializing stake table from node for epoch {epoch:?}");
+
+    match epoch {
+        Some(epoch) => loop {
+            match surf_disco::Client::<tide_disco::error::ServerError, StaticVersion<0, 1>>::new(
+                sequencer_url.clone(),
+            )
+            .get::<Vec<PeerConfig<SeqTypes>>>(&format!("node/stake-table/{}", epoch.u64()))
+            .send()
+            .await
+            {
+                Ok(resp) => break Ok(resp),
+                Err(e) => {
+                    tracing::error!("Failed to fetch the network config: {e}");
```

Comment:
```suggestion
                    tracing::error!("Failed to fetch the stake table: {e}");
```

This isn't from this PR but since the code was moved, might as well fix it now.